src/OpenTracing/Propagation/BinaryCarrier.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace OpenTracing.Propagation
+{
+    /// <summary>
+    /// A <see cref="ITextMapCarrier"/> which allows a byte array to be used as a carrier object.
+    /// </summary>
+    public class BinaryCarrier : IInjectCarrier, IExtractCarrier
```

Comment:
This is a matter of taste, but I think it's fine to collapse `IInjectCarrier` and `IExtractCarrier` into plain-old `ICarrier`... slightly less flexible/clean, but slightly fewer concepts for new users.